### PR TITLE
Add button to swap custom min and custom max

### PR DESCRIPTION
**Pull request by axelboc**
The button only appears if both min an max have custom values (i.e. only if the user purposefully entered two custom values in the wrong order):

image

Swapping resolves the situation:

image

If any bound is set to auto-scale, the button is not visible:

image

### CODE REVIEW

File: src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      <span>
+        Min greater than max
+        <br />
+        <FiCornerDownRight /> falling back to <strong>data range</strong>
+      </span>
+      {!isAuto && (
+        <button
+          className={styles.actionBtn}
+          type="button"
+          onClick={() => onSwap()}
```
Comment (loichuder) [inline src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx:24]: Missing some `aria` attributes ?

**Review by loichuder — CHANGES_REQUESTED** [review]

File: src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx (hunk 1 above)
Comment (axelboc) [reply src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx:24]: The icon inside the button has a label, so it's all good 👍

File: src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+interface Props {
+  isAuto: boolean;
+  onSwap: () => void;
+}
+
+function MinGreaterErrorMessage(props: Props) {
+  const { isAuto, onSwap } = props;
+
+  return (
+    <p className={styles.minMaxError}>
```
[the remark this answers is not in the document]
Comment (axelboc) [reply src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx:14]: Completely forgot about it, thanks! I'm abstracting the behaviour in a `StickyError` component. No need for error-specific components any more.

[the remark this answers is not in the document]
Comment (axelboc) [reply src/h5web/toolbar/controls/DomainSlider/MinGreaterErrorMessage.tsx]: Scratch that, as discussed, I've replaced the sticky behaviour with invisible padding around the tooltip. I still abstracted the error messages with a new component called `ErrorMessage`.

File: src/h5web/toolbar/controls/DomainSlider/ErrorMessage.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+interface Props {
+  message: string;
+  fallback: string;
+  showSwapBtn?: boolean;
+  onSwap?: () => void;
+}
+
+function ErrorMessage(props: Props) {
+  const { message, fallback, showSwapBtn = false, onSwap } = props;
```
Comment (loichuder) [inline src/h5web/toolbar/controls/DomainSlider/ErrorMessage.tsx:13]: I would send the `error` as props and put the call of `getBoundErrorProps` (that would be renamed) in this component.

For `minGreater`, you could add a new type of `BoundError` created at https://github.com/silx-kit/h5web/pull/568/files#diff-0bff5f3cf4d4a01448d5c83c2021ad2d214a71511c758e866be6bad0698205aeR49. You would then need to add the appropriate associated message in `getBoundErrorProps`.

But I won't push strongly for this, feel free to keep as it is.

**Review by loichuder — APPROVED** [review]

File: src/h5web/toolbar/controls/DomainSlider/ErrorMessage.tsx (hunk 3 above)
Comment (axelboc) [reply src/h5web/toolbar/controls/DomainSlider/ErrorMessage.tsx:13]: No, you're right, great idea!

**Pull request merged**